### PR TITLE
Add :env option when defining roles

### PR DESCRIPTION
**Pull request by brienw**
Addresses [#213]

### CODE REVIEW

**Comment by brienw** [comment]
I decided to just piggy back on top of SSHKit Context environments. Let me know what you guys thing. This will set all provided `:env` variables for all remote commands

File: test/bootleg/config_test.exs [hunk 1]
```diff
@@ -91,6 +91,31 @@ defmodule Bootleg.ConfigTest do
...
+
+    role :build, "build.labzero.com", user: "brien", env: %{FOO: "1234"}
+    assert roles() == [
+      build: %Bootleg.Role{
+        hosts: [
+          %Bootleg.Host{
+            host: %SSHKit.Host{name: "build.labzero.com", options: [user: "brien"]},
+            options: [user: "brien", env: %{FOO: "1234"}]
+          }
+        ],
```
Comment (pmeinhardt) [inline test/bootleg/config_test.exs]: I haven't had a chance yet to study the codebase in greater detail. I was just wondering whether the implementation could be simplified by building an `SSHKit.Context` directly? À la:

```elixir
assert roles() == [
  build: %Bootleg.Role{
    contexts: [
      %SSHKit.Context{…}
    ],
    name: build,
    options: […]
  }
]
```

At the moment, I see you have another indirection through `Bootleg.Host` which looks like it stores the same values as an `SSHKit.Context`. But maybe I'm wrong.

This could be something for a separate PR?

Comment (brienw) [reply test/bootleg/config_test.exs]: Definitely worth discussing more as a separate issue/PR. The original intention of having an abstraction for role was partly due to it being a moving target and us not wanting to go too far down a purely SSHKit.Context route only to hit a snag with a future feature where we may have needed extra behavior. It's been long enough that I don't fully recall the full reasoning, but now that bootleg has has some time to mature and settle down a bit, it may be worth taking a fresh look to see if we still want/need the `Bootleg.Role` module
